### PR TITLE
detect-detection: Convert unittests to new FAIL/PASS API

### PR DESCRIPTION
**Pull request by hsadia538**
Make sure these boxes are signed before submitting your Pull Request -- thank you.

- [X] I have read the contributing guide lines at https://redmine.openinfosecfoundation.org/projects/suricata/wiki/Contributing
- [X] I have signed the Open Information Security Foundation contribution agreement at https://suricata.io/about/contribution-agreement/
- [ ] I have updated the user guide (in doc/userguide/) to reflect the changes made (if applicable)

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:

Previous PR: https://github.com/OISF/suricata/pull/8277

Describe changes:

- Add commit for clang checks

### CODE REVIEW

File: src/detect-detection-filter.c [hunk 1]
```diff
@@ -302,14 +302,10 @@ static int DetectDetectionFilterTestParse01 (void)
...
 static int DetectDetectionFilterTestParse02 (void)
 {
-    DetectThresholdData *df = NULL;
-    df = DetectDetectionFilterParse("track both,count 10,seconds 60");
-    if (df && (df->track == TRACK_DST || df->track == TRACK_SRC) && (df->count == 10) && (df->seconds == 60)) {
-        DetectDetectionFilterFree(NULL, df);
-        return 0;
-    }
+    DetectThresholdData *df = DetectDetectionFilterParse("track both,count 10,seconds 60");
+    FAIL_IF_NOT_NULL(df);
```
Comment (jufajardini) [inline src/detect-detection-filter.c:310]: Aren't we missing some FAIL checks, here?

File: src/detect-detection-filter.c [hunk 2]
```diff
@@ -375,14 +368,10 @@ static int DetectDetectionFilterTestParse05 (void)
  */
 static int DetectDetectionFilterTestParse06 (void)
 {
-    DetectThresholdData *df = NULL;
-    df = DetectDetectionFilterParse("count 10, track by_dst, seconds 0");
-    if (df && (df->track == TRACK_DST) && (df->count == 10) && (df->seconds == 0)) {
-        DetectDetectionFilterFree(NULL, df);
-        return 0;
-    }
```
Comment (jufajardini) [inline src/detect-detection-filter.c:383]: Same as prior comment

File: src/detect-detection-filter.c [hunk 3]
```diff
@@ -481,70 +454,61 @@ static int DetectDetectionFilterTestSig2(void)
...
     SigMatchSignatures(&th_v, de_ctx, det_ctx, p);
-    alerts += PacketAlertCheck(p, 10);
+    FAIL_IF(PacketAlertCheck(p, 10));
 
-    if (alerts == 0)
-        result = 1;
-
-    SigGroupCleanup(de_ctx);
-    SigCleanSignatures(de_ctx);
+    // FAIL_IF_NOT(alerts == 0);
```
Comment (jufajardini) [inline src/detect-detection-filter.c:500]: Leftover commented out code :P

File: src/detect-detection-filter.c [hunk 4]
```diff
@@ -554,81 +518,71 @@ static int DetectDetectionFilterTestSig3(void)
...
-        result = 1;
-    else {
-        if (alerts != 3)
-            printf("alerts: %d != 3: ", alerts);
-        if (drops != 3)
-            printf("drops: %d != 3: ", drops);
-    }
-
     SigGroupCleanup(de_ctx);
     SigCleanSignatures(de_ctx);
```
Comment (jufajardini) [inline src/detect-detection-filter.c:583]: These can be removed from this test and the others that have used `DetectEngineAppendSig`

**Review by jufajardini — CHANGES_REQUESTED** [review]
Almost there! :)

Left a few inline comments, but the most complicated parts are working, from what I could see.

I think that with this final clean up/conversion, you can remove the 
```
 *  \retval 1 on succces
 *  \retval 0 on failure
 ```
 
Portion of the comment on the tests that have been converted to the new FAIL/PASS macros :)

File: src/detect-detection-filter.c (hunk 1 above)
Comment (jufajardini) [reply src/detect-detection-filter.c:310]: I stand corrected :P https://github.com/OISF/suricata/pull/8088#discussion_r1004861760

**Comment by hsadia538** [comment]
Followed by #8283

**Pull request closed without merging**